providers/amazon/src/airflow/providers/amazon/aws/executors/aws_lambda/lambda_executor.py

```diff
@@ -0,0 +1,488 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+import time
+from collections import deque
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from boto3.session import NoCredentialsError
+from botocore.utils import ClientError
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.executors.base_executor import BaseExecutor
+from airflow.models.taskinstancekey import TaskInstanceKey
+from airflow.providers.amazon.aws.executors.aws_lambda.utils import (
+    CONFIG_GROUP_NAME,
+    INVALID_CREDENTIALS_EXCEPTIONS,
+    AllLambdaConfigKeys,
+    CommandType,
+    LambdaQueuedTask,
+)
+from airflow.providers.amazon.aws.executors.utils.exponential_backoff_retry import (
+    calculate_next_attempt_delay,
+    exponential_backoff_retry,
+)
+from airflow.providers.amazon.aws.hooks.lambda_function import LambdaHook
+from airflow.providers.amazon.aws.hooks.sqs import SqsHook
+from airflow.stats import Stats
+from airflow.utils import timezone
+
+from tests_common.test_utils.version_compat import AIRFLOW_V_3_0_PLUS
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+    from airflow.executors import workloads
+    from airflow.models.taskinstance import TaskInstance
+
+
+class AwsLambdaExecutor(BaseExecutor):
+    """
+    An Airflow Executor that submits tasks to AWS Lambda asynchronously.
+
+    When execute_async() is called, the executor invokes a specified AWS Lambda function (asynchronously)
+    with a payload that includes the task command and a unique task key.
+
+    The Lambda function writes its result directly to an SQS queue, which is then polled by this executor
+    to update task state in Airflow.
+    """
+
+    # Maximum number of retries to invoke Lambda.
+    MAX_INVOKE_ATTEMPTS = conf.get(
+        CONFIG_GROUP_NAME,
+        AllLambdaConfigKeys.MAX_INVOKE_ATTEMPTS,
+        fallback=3,
```

Comment:
It applies to call `conf.get`, the `fallback` should be necessary. The default is already defined in config

Reply:
This one is weird, I thought so too, but my testing has shown that this does not work as expected? Maybe it's something on my end?

Reply:
Did you test it after you run `pre-commits`? `pre-commits` generate `get_provider_info.py` which are used to get configuration for providers. So you must run pre-commits to generate these file and then after these configuration are getting picked-up.

Another idea would be, `@providers_configuration_loaded` is missing. This decorator loads the configuration from providers. Since the executor is instantiated in the scheduler component, we do not load configuration from providers in scheduler?

Reply:
Interesting, I'll go have a look at these two options!